src/wallet/rpcdump.cpp

```diff
@@ -660,19 +660,26 @@ UniValue dumpwallet(const JSONRPCRequest& request)
         std::string strAddr = CBitcoinAddress(keyid).ToString();
         CKey key;
         if (pwallet->GetKey(keyid, key)) {
+            CKeyMetadata meta;
```

Comment:
In this case how about:
```cpp
CKeyMetadata meta = pwallet->mapKeyMetadata[keyid];
```

Reply:
`operator[]` can also create map items. Even if this never occurs, it needs to generate code for it.

Reply:
Right, off topic, but if above was changed too `const CWallet * const pwallet` then that would be false.

Reply:
`operator[]` isn't valid at all on const maps AFAIK...?

Reply:
Indeed! Sorry for the noise.

Reply:
n.b. C++11 added `std::map::at`, but this isn't useful since we need to correctly handle the case where there is no metadata.